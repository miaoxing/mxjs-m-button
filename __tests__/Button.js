import Button from '..';
import {render} from '@testing-library/react';

describe('Button', () => {
  test('basic', () => {
    const {container} = render(<Button>test</Button>);
    expect(container).toMatchSnapshot();
  });

  test('variant: primary', () => {
    const {container} = render(<Button variant="primary">primary</Button>);
    expect(container).toMatchSnapshot();
  });

  test('variant: secondary', () => {
    const {container} = render(<Button variant="secondary">secondary</Button>);
    expect(container).toMatchSnapshot();
  });
});
