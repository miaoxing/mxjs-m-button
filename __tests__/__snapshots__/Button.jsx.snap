// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button basic 1`] = `
<div>
  <taro-button-core
    class="mx-btn"
  >
    test
  </taro-button-core>
</div>
`;

exports[`Button href 1`] = `
<div>
  <taro-button-core
    class="mx-btn"
  >
    secondary
  </taro-button-core>
</div>
`;

exports[`Button href 2`] = `
Array [
  Array [
    Object {
      "url": "next?id=1",
    },
  ],
]
`;

exports[`Button variant: primary 1`] = `
<div>
  <taro-button-core
    class="mx-btn mx-btn-primary"
  >
    primary
  </taro-button-core>
</div>
`;

exports[`Button variant: secondary 1`] = `
<div>
  <taro-button-core
    class="mx-btn mx-btn-secondary"
  >
    secondary
  </taro-button-core>
</div>
`;
